### PR TITLE
Playbook for integration with external docker registry

### PR DESCRIPTION
**Pull request by sabre1041**
Playbook for integration with external docker registry

### CODE REVIEW

**Comment by etsauer** [comment]
@sabre1041 the technical content of this guide looks good and accurate. I would like, however for the doc to be a bit more opinionated, and make a recommendation on what we would suggest as an overall workflow that would involve an external registry.

**Comment by etsauer** [comment]
@JaredBurck please review as well

**Comment by sabre1041** [comment]
@etsauer updates have been made. please review

File: playbooks/operationalizing/external-docker-registry-integration.adoc [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+
+To configure docker to trust the certificate of a remote repository that is not included in the platform trust store, create a folder within the `/etc/docker/certs.d` folder with the name and optional port value. If the remote registry had the address of `external-registry.example.com:5000` the folder path would be `/etc/docker/certs.d/external-registry.example.com`.
+
+Copy the certificate into this folder and rename the file *ca.crt*
+
+=== Insecure Registry
+
+Even though Docker and OpenShift emphasize secure communication, the Docker daemon can be configured to retrieve images from insecure registries. Insecure registries can refer to ones that are protected with self signed or untrusted SSL certificates or ones that do not communicate over secure channels (HTTP).
```
Comment (sabre1041) [inline playbooks/operationalizing/external-docker-registry-integration.adoc]: The last part of the final sentence describes the use of an insecure communication channel (ie HTTP). Existing content should be valid

**Comment by raffian** [comment]
We have a Docker repo in Artifactory and want to integrate Openshift to pull/push from that repo, avoiding the internal registry all together. Are these instructions applicable to Openshift 3.6?

**Comment by etsauer** [comment]
@raffian they should be relevant for 3.6 still. If you come across any issues, please open an issue and we'll get it fixed.

**Comment by raffian** [comment]
@etsauer Ok, sounds good, although references to `DockerImage` appear out of place; there is no such object type in 3.6, is this section still valid?

https://github.com/redhat-cop/openshift-playbooks/blob/master/playbooks/continuous_delivery/external-docker-registry-integration.adoc#storing-images-as-a-result-of-a-build

**Comment by sabre1041** [comment]
@raffian The `DockerImage` type is still valid for OpenShift 3.6 as described in the following documentation

https://docs.openshift.com/container-platform/3.6/dev_guide/builds/build_output.html

**Comment by raffian** [comment]
@sabre1041 ahhh, I used `oc types` as my reference; thanks for that link. Can we use API tokens issued from Artifactory in place of literal passwords when setting up secrets for secure registeries?

**Comment by raffian** [comment]
@etsauer  We got this working on 3.6, but had some trouble assigning docker secret to builder service accounts, partly because the section detailing steps for binding secrets to service accounts -- [Accessing Secure Registrises](http://v1.uncontained.io/playbooks/continuous_delivery/external-docker-registry-integration.html#accessing-secure-registrises) -- is a bit unclear, especially to beginners. The docs should be explicit about which build accounts get the secret since it's reasonable to expect separate builder service accounts exist per namespace. I would also suggest adding the oc command to validate the service account actually has the secret necessary for accessing the external registry:

```
#do this per namespace 
oc describe sa builder -n <ns>

# should see 'external-registry secret' for pull and mountable
Image pull secrets:     builder-hello-cnfd6
                                 external-registry

Mountable secrets:      builder-token-hf5we
                                   builder-dockercfg-98ugd
                                   external-registry
```

**Comment by etsauer** [comment]
@raffian thanks for this feedback. Let's move this conversation into https://github.com/redhat-cop/openshift-playbooks/issues/214 so we can track progress on a docs update.

**Pull request merged**